distribution/docker/build.gradle

```diff
@@ -91,20 +98,79 @@ void addCopyDockerfileTask(final boolean oss) {
   }
 }
 
+Closure createAndSetWritable = {location ->
+  File file = file(location);
+  file.mkdirs()
+  file.setWritable(true, false)
+}
+
+task copyKeystore(type: Sync) {
+  from project(':x-pack:plugin:core')
+          .file('src/test/resources/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode.jks')
+  into "${buildDir}/certs"
+  doLast {
+    file("${buildDir}/certs").setReadable(true, false)
+    file("${buildDir}/certs/testnode.jks").setReadable(true, false)
+  }
+}
+
 preProcessFixture {
   dependsOn taskName("copy", true, "DockerContext")
   dependsOn taskName("copy", true, "Dockerfile")
   dependsOn taskName("copy", false, "DockerContext")
   dependsOn taskName("copy", false, "Dockerfile")
+  dependsOn copyKeystore
+  doLast {
+    // tests expect to have an empty repo
+    project.delete("${buildDir}/repo")
+    project.delete("${buildDir}/oss-repo")
+    createAndSetWritable("${buildDir}/repo")
+    createAndSetWritable("${buildDir}/oss-repo")
+    createAndSetWritable("${buildDir}/logs/default-1")
+    createAndSetWritable("${buildDir}/logs/default-2")
+    createAndSetWritable("${buildDir}/logs/oss-1")
+    createAndSetWritable("${buildDir}/logs/oss-2")
+  }
 }
 
-postProcessFixture.doLast {
-  println "docker default distro is on port: ${ext."test.fixtures.elasticsearch-default.tcp.9200"}, " +
-          "oss is on: ${ext."test.fixtures.elasticsearch-oss.tcp.9200"}"
+configurations {
```

Comment:
Point of personal preference: I like to define all configuration and dependencies in one place at the top of the build script, even if they are used in later logic. It just makes it easier to reason about dependency management stuff IMO.